Ethereum/README.md

```diff
@@ -1,15 +1,174 @@
-# StraitsX-tokens
+# StraitsX-tokens (XSGD)
+
+## Token Description
+The StraitsX SGD (XSGD) token is the first digital token fully collateralized one-for-one with Singapore Dollar (SGD) and representing 1 SGD. The XSGD token is written as a smart contract on the Ethereum blockchain following the ERC20 standard and is governed by the StraitsX network whereby XSGD tokens are centrally minted and burned by Xfers Pte. Ltd. 
+
+The Xfers payment service is regarded as a Widely Accepted Stored Value Facility under Singapore law. Xfers Pte. Ltd., is the Approved Holder of the Xfers Wallet Stored Value Facility.
 
-Fiat tokens on the [StraitsX Project](https://www.xfers.com/sg/stablecoin/).
+https://www.xfers.com/
 
-# Potential signature pitfalls
+## Smart Contract Specifications
+The XSGD contract has been written by the StraitsX team to fit both the specific needs of the XSGD token as described in the StraitsX Whitepaper and the requirement to comply with local regulation.
+
+The XSGD contract consists of two communicating contracts:
+- [token contract](https://github.com/Xfers/StraitsX-tokens/blob/master/Ethereum/contracts/FiatTokenV1.sol)
+- [proxy contract](https://github.com/Xfers/StraitsX-tokens/blob/master/Ethereum/contracts/FiatTokenProxy.sol)
+
+The token contract represents a standard fungible token contract with minting and burning features, while an [unstructured storage proxy pattern](https://docs.openzeppelin.com/upgrades-plugins/1.x/proxies#unstructured-storage-proxies) is used for the proxy contract. This pattern enables the proxy to use the logic of the token contract but modifying its own storage state. The proxy can point to the new contract address should there be upgrades to the token implementation in the future.
```

Comment:
Is that a typo? 🤔 `but` -> `by`

Reply:
nope, ok will rephrase to make clearer. 

"This pattern enables the token contract to modify the storage of the proxy contract. Should there be an upgrade to the token implementation, the proxy can point to the new contract without having to transfer the storage state since it resides on the proxy."

Reply:
👍